### PR TITLE
Refactor overloaded type_name query parameter to type/category

### PR DESCRIPTION
**Pull request by patrickvinograd**
Fixes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1814
requires corresponding front-end fix to use updated facets and query parameters. 

New parameters are:
`category` = school/employer
`type` = private/for profit/ojt/foreign/etc

That's both what should be sent as query parameters, and what will come back in the facets object for populating the search form.

### CODE REVIEW

File: app/models/institution.rb [hunk 1]
```diff
@@ -221,17 +221,22 @@ def self.autocomplete(search_term, limit = 6)
...
-    case value
-    when 'true', 'yes'
-      where(field => true)
-    when 'false', 'no'
-      where.not(field => true)
-    when 'school' # field: institution_type_name
-      where.not(field => EMPLOYER)
-    when 'employer' # field: institution_type_name
-      where(field => EMPLOYER)
+    if field == :category
```
Comment (patrickvinograd) [inline app/models/institution.rb]: category is the only real "synthetic" query parameter that doesn't map straight through to a database value, so for clarity's sake I pulled it out of the single case statement that was here before.

**Review by U-DON — APPROVED** [review]
LGTM.

**Comment by saneshark** [comment]
So correct me if I'm wrong but with category having its own if / else this means that it will take precedence over the other attributes. To me it seems more logical to have the filters update / refresh on the front-end in a dynamic manner, this way both have equal precedence vs one having greater precedence than the other. Not only would this simplify some of the logic on the backend for institution_type_name column, but it also makes for a more dynamic set of filters for veterans, does it not?

**Comment by patrickvinograd** [comment]
No this does not affect precedence. The :filter scope is just turning query params into the where clauses of the SQL statement, all of which are ANDed together.

The old code was taking a shortcut by saying if `value == school`, generate `where institution_type_name != OJT`, without actually checking whether the query parameter in question was `type_name`. So in practice you could submit an API request with `?state=school` and it would generate sql like `where state != OJT`. Not particularly dangerous, but incorrect. The version in this PR makes the behavior more explicit by only treating "school" and "employer" as meaningful query parameter values when the query parameter = "category".

**Review by saneshark — APPROVED** [review]

**Pull request merged**